content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -333,6 +341,166 @@ The `olm.gvk.required` property [cue][cuelang-spec] schema is:
 }
 ```
 
+#### `olm.constraint`
+
+An [`olm.constraint` property][generic-constraint-ep] defines a dependency constraint of a particular type.
+The `message` field is recommended but not required to be populated so readers know why a constraint
+was specified, and errors can contain this string. The supported types are detailed in subsections.
+
+The `olm.constraint` property [cue][cuelang-spec] schema is:
+
+```cue
+#ConstraintValue: {
+  message?: string
+
+  { gvk: #GVK } |
+  { package: #Package } |
+  { cue: #Cue } |
+  { all: null | #CompoundConstraintValue } |
+  { any: null | #CompoundConstraintValue } |
+  { none: null | #CompoundConstraintValue }
+}
+
+#PropertyOLMConstraint: {
+  type: "olm.constraint"
+  value: #ConstraintValue
+}
+```
+
+
+##### `#GVK`
+
+`#GVK` is identical to the old top-level constraint `olm.gvk.required` value. The [cue][cuelang-spec] schema is:
+
+```cue
+#GVK: {
+  group: string & !=""
+  version: string & !=""
+  kind: string & !=""
+}
+```
+
+Example:
+
+```cue
+#PropertyOLMConstraint & {
+  value: {
+    message: "required for ..."
+    gvk: #GVK & {
+      group: "example.com"
+      version: "v1"
+      kind: "Foo"
+    }
+  }
+}
+```
+
+##### `#Package`
+
+`#Package` is identical to the old top-level constraint `olm.package.required` value. The [cue][cuelang-spec] schema is:
+
+```cue
+#Package: {
+  packageName: string & !=""
+  versionRange: string & !=""
+}
+```
+
+Example:
+
+```cue
+#PropertyOLMConstraint & {
+  value: {
+    message: "required for ..."
+    package: #Package & {
+      packageName: "foo"
+      versionRange: ">=1.0.0"
+    }
+  }
+}
+```
+
+##### `#Cue`
+
+TODO
```

Comment:
@dinhxuanvu this needs a follow-up with a cue schema and short description with link to the EP.